frontend/src/lib/colors.ts

```diff
@@ -97,3 +97,11 @@ export function getChartColors(backgroundColor: string): string[] {
 
     return colorsForBackground(colors[0], colors[1], colors[2])
 }
+
+export function setOpacityToHexColor(hexCode: string, opacity: number): string {
+    // Alpha can be added to hex values to obtain a lighter set of colors
+    // 1A as suffix means the alpha is set to 10% or opacity is 0.1
+    const alpha = Math.floor(255 * opacity).toString(16)
+    const hexWithAlpha = `${hexCode}${alpha}`
+    return hexWithAlpha
```

Comment:
Hi, this function fails some cases:

```javascript
// 3-letter codes
setOpacityToHexColor("#fff", 0.4) === "#fff66"

// tiny numbers
setOpacityToHexColor("#ffffff", 0.1)  === "ffffff19"
setOpacityToHexColor("#ffffff", 0.01) === "ffffff2"
```

Reply:
Since a lot of things have changed since, would it make sense to raise a new PR for this?